### PR TITLE
Create a no-opi-deps flag for module.ini

### PR DESCRIPTION
**Pull request by mfurseman**
Sometimes configure/RELEASE specifies build time dependencies that are not required for displays. CS-Studio was still including these via the runcss.sh file and generating errors when starting up if the *App/opi/opi directory doesn't exist. By specifying a list of dependencies which are not GUI dependencies which can filter these from the runcss.sh.

### CODE REVIEW

File: dls_css_utils/dependency.py [hunk 1]
```diff
@@ -119,15 +119,13 @@ def find_dependencies(self):
...
                 if is_valid(dependency, self._module_path):
+                    module_coord = coordinates.from_path(dependency.path)
                     try:
                         module_cfg = config.parse_module_config(dependency.path)
                         name = config.module_name(module_cfg)
                     except config.ConfigError:
-                        name = dependency.name
-
-                    # exclude THIS module from the dependency list
-                    if not name == self._module_name:
```
Comment (nickbattam) [inline dls_css_utils/dependency.py]: Can you safely remove this piece of functionality? THIS being a dependency caused a lot if problems in earlier iterations

File: dls_css_utils/run_script.py [hunk 2]
```diff
@@ -155,6 +174,7 @@ def generate_from_root(root):
     # now we need to find dependencies from root/configure/RELEASE
     dependencies = dependency.DependencyParser.from_path(root, module_name, additional_depends=extra_deps)
     deps = dependencies.find_dependencies()
+    deps = _filter_deps(deps, no_opi_deps)
```
Comment (nickbattam) [inline dls_css_utils/run_script.py]: reassigning this dict is inelegant; you could combine the _filter_deps call with the previous or subsequent line, or give it a new variable name 'filtered_deps'

File: test/test_dependency.py [hunk 3]
```diff
@@ -108,6 +108,30 @@ def test_returns_empty_list_if_no_depends_from_path(self):
         depends = d.find_dependencies()
         self.assertDictEqual({}, depends)
 
+    def test_dependencies_returns_correct_depends_for_alphanumeric_version_name(self):
+        path = "/dls_sw/prod/R3.14.12.3/ioc/LI/TI/5-9"
```
Comment (nickbattam) [inline test/test_dependency.py]: this is a fragile test as it depends on the file system

File: test/test_dependency.py [hunk 4]
```diff
@@ -108,6 +108,30 @@ def test_returns_empty_list_if_no_depends_from_path(self):
...
+                coordinates.create('/dls_sw/prod/R3.14.12.3', 'support', 'mrfTiming', '2-4-1dls9'),
+            'TimingTemplates':
+                coordinates.create('/dls_sw/prod/R3.14.12.3', 'support', 'TimingTemplates', '6-10'),
+            'LinacGunTX':
+                coordinates.create('/dls_sw/prod/R3.14.12.3', 'support', 'LinacGunTX', '3-0'),
+            'calc':
+                coordinates.create('/dls_sw/prod/R3.14.12.3', 'support', 'calc', '3-1'),
+                }
+        self.assertDictEqual(expected, depends)
+
```
Comment (nickbattam) [inline test/test_dependency.py]: You need to add more tests to confirm that filter_deps and no_opi_depends code are behaving as intended. As these both take all their data as arguments you can construct strong test cases that cover all the edge cases.

File: dls_css_utils/dependency.py (hunk 1 above)
Comment (mfurseman) [reply dls_css_utils/dependency.py]: The `is_valid` check is checking for `dependency.path != self._module_path`. This seems like it ought to catch all those cases.
